src/main/java/com/agency04/sbss/pizza/service/impl/util/converter/DeliveryOrderFormToCustomerConverter.java

```diff
@@ -11,18 +12,18 @@
 @Service
 public class DeliveryOrderFormToCustomerConverter implements Converter<DeliveryOrderForm, Customer> {
 
+	private final CustomerRepository customerRepository;
+
+	public DeliveryOrderFormToCustomerConverter(CustomerRepository customerRepository) {
+		this.customerRepository = customerRepository;
+	}
+
 	@Override
 	public Customer convert(DeliveryOrderForm deliveryOrderForm) {
 
-		if (deliveryOrderForm.getCustomer() == null
-				|| deliveryOrderForm.getCustomer().getUsername() == null
-				|| deliveryOrderForm.getCustomer().getAddress() == null
-				|| deliveryOrderForm.getCustomer().getPhone() == null)
+		if (deliveryOrderForm.getCustomerUsername() == null)
 			throw new NullPointerException("Customer personal data missing.");
 
-		return new Customer(
-				deliveryOrderForm.getCustomer().getUsername(),
-				deliveryOrderForm.getCustomer().getAddress(),
-				deliveryOrderForm.getCustomer().getPhone());
+		return customerRepository.findById(deliveryOrderForm.getCustomerUsername()).get();
```

Comment:
Zašto se dohvaća objekt iz Optionala bez prethodne provjere s isPresent()? Što bi bilo očekivano ponašanje API-ja u tom slučaju? Recimo kreirati custom RuntimeException pa njega baciti s "orElseThrow()" i hendlati to u "ControllerAdviceu". NoSuchElementException sigurno nije dobro rješenje.